mycroft/util/file_utils.py

```diff
@@ -275,3 +275,28 @@ def create_file(filename):
     ensure_directory_exists(os.path.dirname(filename), permissions=0o775)
     with open(filename, 'w') as f:
         f.write('')
+
+
+def get_temp_path(*args):
+    """Generate a valid path in the system temp directory.
+
+    This method accepts one or more strings as arguments. The arguments are
+    joined and returned as a complete path inside the systems temp directory.
+    Importantly, this will not create any directories or files.
+
+    Example usage: get_temp_path('mycroft', 'audio', 'example.wav')
+    Will return the equivalent of: '/tmp/mycroft/audio/example.wav'
+
+    Args:
+        path_element (str): directories and/or filename
+
+    Returns:
+        (str) a valid path in the systems temp directory
+    """
+    try:
+        path = os.path.join(tempfile.gettempdir(), *args)
+    except TypeError:
+        path = None
+        LOG.error('Could not create a temp path, '
+                  'get_temp_path() only accepts Strings')
```

Comment:
Possibly we should re-raise the exception since the result won't be usable?

Otherwise we should probably add to the docstring that it will return `None` if the folder couldn't be created.

Reply:
Yeah good call - explicitly failing is much better